src/apps/frontend/services/auth.service.ts

```diff
@@ -22,4 +22,24 @@ export default class AuthService extends APIService {
     username,
     password,
   });
+
+  sendOTP = async (
+    phoneNumber: PhoneNumber,
+  ): Promise<ApiResponse<void>> => this.apiClient.post('/accounts', {
+    phone_number: {
+      country_code: phoneNumber.countryCode,
+      phone_number: phoneNumber.phoneNumber,
+    },
+  });
+
+  verifyOTP = async (
+    phoneNumber: PhoneNumber,
+    otp: string,
+  ): Promise<ApiResponse<AccessToken>> => this.apiClient.post('/access-tokens', {
```

Comment:
I think this would return the json inside the api response like `Promise<ApiResponse<JSON(a json object)>>`. We should create an instance of `AccessToken` from the response and return it. I have the same in the password login flow too, let's fix there too if we can.

Reply:
I agree that it should return an object of `ApiResponse` but as per current structure the async hook will need a workaround to make the above statement working properly.
Will create a ticket regarding this and share the link here.

Reply:
Issue link: https://github.com/jalantechnologies/rflask-boilerplate/issues/54